net/core/dev.c

```diff
@@ -157,6 +157,10 @@
 
 static DEFINE_SPINLOCK(ptype_lock);
 static DEFINE_SPINLOCK(offload_lock);
+#ifdef CONFIG_XDP_LUA
+#include <net/xdplua.h>
+#undef free
```

Comment:
Since Lunatik is defining `free`: https://github.com/luainkernel/lunatik/blob/master/lua/luaconf.h#L875 it also substitutes `free` in the following line https://github.com/luainkernel/linux/blob/master/net/core/dev.c#L5548, which then raises an error, since there is no `kfree` in the struct. In hindsight, I think it would be better to put this in `xdplua.h`, even tough `dev.c` is the only file where such a problem occurs.

Reply:
hm.. not sure.. it might be better to handle this in lunatik's luaconf.h..